src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3663,6 +3645,9 @@ internal void SetToConnected()
             // some point in time update the perf counter as well.
             _isConnected = true;
             _isDisconnected = false;
+            _pollPendingConnect = false;
+            _rightEndPoint ??= _pendingConnectRightEndPoint;
```

Comment:
If I understand correctly, _pendingConnectRightEndPoint should never be null here, correct? Should we assert that?

Reply:
It's `null` when called for an accepted socket.